### PR TITLE
BigTable: simplify configuration of data clients

### PR DESCRIPTION
**Pull request by adutra**
Note: this will conflict with #7636 but I'll update the PR if that one gets merged first.

### CODE REVIEW

File: servers/quarkus-common/src/main/java/org/projectnessie/quarkus/config/QuarkusBigTableConfig.java [hunk 1]
```diff
@@ -18,48 +18,15 @@
...
-import java.time.Duration;
-import java.util.Map;
-import java.util.Optional;
-import java.util.OptionalInt;
+import org.projectnessie.versioned.storage.bigtable.BigTableBaseBackendConfig;
 
 @StaticInitSafe
 @ConfigMapping(prefix = "nessie.version.store.persist.bigtable")
-public interface QuarkusBigTableConfig {
+public interface QuarkusBigTableConfig extends BigTableBaseBackendConfig {
```
Comment (dimas-b) [inline servers/quarkus-common/src/main/java/org/projectnessie/quarkus/config/QuarkusBigTableConfig.java:25]: My thought exactly :)

File: versioned/storage/bigtable/src/main/java/org/projectnessie/versioned/storage/bigtable/BigTableBaseBackendConfig.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+  Optional<String> quotaProjectId();
+
+  Optional<String> endpoint();
+
+  Optional<String> mtlsEndpoint();
+
+  Optional<String> emulatorHost();
+
+  Optional<String> tablePrefix();
```
Comment (dimas-b) [inline versioned/storage/bigtable/src/main/java/org/projectnessie/versioned/storage/bigtable/BigTableBaseBackendConfig.java:39]: Aside from `tablePrefix()` these settings look like BT client settings. `BigTableBackend` does not appear to use them.

WDYT about keeping `tablePrefix()` separate and renaming this to `BigTableClientConfig`?

Comment (snazy) [reply versioned/storage/bigtable/src/main/java/org/projectnessie/versioned/storage/bigtable/BigTableBaseBackendConfig.java:39]: > renaming this to BigTableClientConfig?

"backend" refers to the Nessie storage backend implementation - not some "BT backend"

**Review by snazy — APPROVED** [review]

File: versioned/storage/bigtable/src/main/java/org/projectnessie/versioned/storage/bigtable/BigTableBackendConfig.java [hunk 3]
```diff
@@ -17,20 +17,17 @@
 
 import com.google.cloud.bigtable.admin.v2.BigtableTableAdminClient;
 import com.google.cloud.bigtable.data.v2.BigtableDataClient;
-import java.util.Optional;
 import javax.annotation.Nullable;
 import org.immutables.value.Value;
 
 @Value.Immutable
-public interface BigTableBackendConfig {
+public interface BigTableBackendConfig extends BigTableBaseBackendConfig {
```
[the remark this answers is not in the document]
Comment (adutra) [reply versioned/storage/bigtable/src/main/java/org/projectnessie/versioned/storage/bigtable/BigTableBackendConfig.java]: Done, let me know what you think.

**Review by dimas-b — APPROVED** [review]
LGTM - thx :)

**Pull request merged**